db/db_impl/db_impl_compaction_flush.cc

```diff
@@ -3962,8 +3963,14 @@ void DBImpl::GetSnapshotContext(
 
 Status DBImpl::WaitForCompact(
     const WaitForCompactOptions& wait_for_compact_options) {
+  if (wait_for_compact_options.flush) {
+    Status s = DBImpl::Flush(FlushOptions());
```

Comment:
I believe that this will only work for default column family. I guess I'd have to use AtomicFlush to include all CFs, please correct me if I am wrong here.

For `AtomicFlushMemTables()`, I'd have to provide `FlushReason`. Is this wait going to be considered `kManualFlush`? Or should I just use `kOthers`?

Reply:
Yes, I think kManualFlush is accurate enough. I recommend calling DBImpl::FlushForGetLiveFiles() to flush all, though maybe refactor/rename that function for different FlushReasons.